e2e/iam/projects_api_keys_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build e2e iam
+
+package iam_test
+
+import (
+	"encoding/json"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/mongodb/mongocli/e2e"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+func TestProjectsAPIKeys(t *testing.T) {
+	cliPath, err := e2e.Bin()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	var ID string
+
+	// This test must run first to grab the ID of the project to later describe
+	t.Run("Create", func(t *testing.T) {
+		cmd := exec.Command(cliPath, iamEntity,
+			projectEntity,
+			apiKeysEntity,
+			"create",
+			"--desc=e2e-test",
+			"--role=ORG_READ_ONLY")
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		a := assert.New(t)
+		if a.NoError(err, resp) {
+			var key mongodbatlas.APIKey
+			if err := json.Unmarshal(resp, &key); err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			a.Equal("e2e-test", key.Desc)
+			ID = key.ID
+		}
+	})
+
+	t.Run("List", func(t *testing.T) {
+		cmd := exec.Command(cliPath, iamEntity, projectEntity, apiKeysEntity, "ls")
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+
+		if err != nil {
+			t.Fatalf("unexpected error: %v, resp: %v", err, string(resp))
+		}
+		var keys []mongodbatlas.APIKey
+		if err := json.Unmarshal(resp, &keys); err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+		assert.NotEmpty(t, keys)
+	})
+
+	t.Run("Describe", func(t *testing.T) {
+		cmd := exec.Command(cliPath, iamEntity, projectEntity, apiKeysEntity, "describe", ID)
```

Comment:
this command doesn't exists, how is this working?

Reply:
I am not sure about this. I run this test in my local machine and I got `nil` as response 🤷‍♂️

Reply:
I tried running `mongocli iam projects apiKeys describe asa` and it doesn't return an exit status of error it just prints usage 🤔  I wonder if we should start validating this because it looks like we don't have tests if we accidentally delete a command

Reply:
yes, I think is a wise thing to do. Maybe instead of validating the output we can just use `assert.NotEmpty `

```
// NotEmpty asserts that the specified object is NOT empty.  I.e. not nil, "", false, 0 or either
// a slice or a channel with len == 0.
```